### DISCUSSION TITLE
Can't figure out how to authorize upload to Supabase.

### DISCUSSION THREAD

**Discussion by ogi-joo** [supabase/supabase#19941] [Questions] [answered]
I created project with **npx create-next-app -e with-supabase**
It comes with login, sign up and homepage which can return user from cookie( )

How do I go about uploading a file to bucket but with users only being able to modify their own folder. 
Server and Client components is what is making the problem, I think.

This is my code:
`
"use client"
import { createClient } from '@supabase/supabase-js';
import { env } from 'process';
import { useState } from 'react';

export default function Notes() {
    const [avatarFile, setAvatarFile] = useState<File | null>(null);
    const [uploading, setUploading] = useState(false);
    const supabase = createClient(process.env.NEXT_PUBLIC_SUPABASE_URL || "", process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY || "");

    const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        if (event.target.files && event.target.files.length > 0) {
            setAvatarFile(event.target.files[0]);
        }
    };

    //get user id from cookies

    const handleUpload = async () => {
        if (avatarFile) {
            setUploading(true);
            try {
                const { data, error } = await supabase.storage.from('videos').upload(avatarFile?.name, avatarFile, {
                    cacheControl: '3600',
                    upsert: false
                });
                if (error) {
                    console.error('Error uploading avatar:', error);
                } else {
                    console.log('Avatar uploaded successfully:', data);
                }
            } catch (error) {
                console.error('Error uploading avatar:', error);
            } finally {
                setUploading(false);
            }
        }
    };

    return (
        <div>
            <input type="file" onChange={handleFileChange} />
            <button onClick={handleUpload} disabled={!avatarFile || uploading}>
                Upload Avatar
            </button>
            {uploading && <p>Uploading...</p>}
        </div>
    );
}`

**Comment by GaryAustin1** [comment, marked as answer]
Assuming you actually have a signed in user session there...
You need to create a Insert policy on that bucket with either the storage UI policies section or using SQL.
https://supabase.com/docs/guides/storage/security/access-control has some examples.
